tests/trainers/test_detection.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+from typing import Any, Callable, Dict, Type, cast
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+from omegaconf import OmegaConf
+from pytorch_lightning import LightningDataModule, Trainer
+
+from torchgeo.datamodules import NASAMarineDebrisDataModule
+from torchgeo.trainers import ObjectDetectionTask
+
+
+class TestObjectDetectionTask:
+    @pytest.mark.parametrize(
+        "name,classname", [("nasa_marine_debris", NASAMarineDebrisDataModule)]
+    )
+    def test_trainer(
+        self, name: str, classname: Type[LightningDataModule], monkeypatch: MonkeyPatch
+    ) -> None:
+        conf = OmegaConf.load(os.path.join("tests", "conf", f"{name}.yaml"))
+        conf_dict = OmegaConf.to_object(conf.experiment)
+        conf_dict = cast(Dict[Any, Dict[Any, Any]], conf_dict)
+
+        # Instantiate datamodule
+        datamodule_kwargs = conf_dict["datamodule"]
+        datamodule = classname(**datamodule_kwargs)
+
+        # Instantiate model
+        model_kwargs = conf_dict["module"]
+        model = ObjectDetectionTask(**model_kwargs)
+
+        # Instantiate trainer
+        trainer = Trainer(fast_dev_run=True, log_every_n_steps=1, max_epochs=1)
+        trainer.fit(model=model, datamodule=datamodule)
+        trainer.test(model=model, datamodule=datamodule)
+
+        if isinstance(getattr(datamodule, "plot"), Callable):  # type: ignore[arg-type]
```

Comment:
Let's put this in a separate test function, there's no reason we need to test this for every datamodule.

Reply:
How do you mean? Add this separately as `test_missing_attribute` because we don't want to mix this with `test_trainer`?

Currently `test_detection.py` only tests the `NasaMarineDebrisDatamodule`.